### PR TITLE
Add no-dns-sec option

### PR DESCRIPTION
**Pull request by fr3dm4n**
This makes it possible to disable dns-sec

### CODE REVIEW

**Comment by lifo101** [comment]
+1 this. I need to disable dnssec in my bind config and don't want to hack the module. Please merge this PR.

File: manifests/server/options.pp [hunk 1]
```diff
@@ -22,6 +22,7 @@
   $check_names_slave = undef,
   $check_names_response = undef,
   $allow_query = [],
+  $no_dns_sec = false,
```
Comment (solarkennedy) [inline manifests/server/options.pp]: What if you named this just "dnssec_validation" and made it default to true? I think that would be easier to read than a double negative.

File: manifests/server/options.pp [hunk 2]
```diff
@@ -22,6 +22,7 @@
...
 ) {
   $valid_check_names = ['fail', 'warn', 'ignore']
```
Comment (solarkennedy) [inline manifests/server/options.pp]: van you add a validate_bool to catch typos?

File: manifests/server/options.pp (hunk 1 above)
Comment (solarkennedy) [inline manifests/server/options.pp]: Can you also add a docstring for this parameter? The style so far is not great, can you start conforming to the styleguide with something like:
```
# == Paramaters
#
# [*dnssec_validation*]
# Bool, defaults to true. If false disabled dnssec and dnssec-validation.
#

File: spec/defines/dns__server__options_spec.rb [hunk 3]
```diff
@@ -111,6 +111,16 @@
     it { should contain_file('/etc/bind/named.conf.options').with_content(/allow-query/)  }
 
   end
+  context 'disable dns-sec option' do
```
Comment (solarkennedy) [inline spec/defines/dns__server__options_spec.rb]: Thank you for writing this!

For style can you add a newline between the contexts and remove the neline on line 122?
(to give each context a more "self-contained" feeling)

**Comment by darkfoxprime** [comment]
Has this issue been taken care of with the recent `dns::server::options::dnssec_validation` parameter?  it takes a value of `yes`, `no`, or `auto`, to set the `dnssec-validation` named option, or `absent` to remove the `dnssec-validation` option completely (for older versions of bind that don't implement dnssec)

**Comment by solarkennedy** [comment]
Correct. dnssec_validation=>absent should fix this.

**Pull request closed without merging**